fs/inode/fs_inode.c

```diff
@@ -138,23 +90,5 @@ int inode_semtake(void)
 
 void inode_semgive(void)
 {
-  DEBUGASSERT(g_inode_sem.holder == getpid());
-
-  /* Is this our last count on the semaphore? */
-
-  if (g_inode_sem.count > 1)
-    {
-      /* No.. just decrement the count */
-
-      g_inode_sem.count--;
-    }
-
-  /* Yes.. then we can really release the semaphore */
-
-  else
-    {
-      g_inode_sem.holder = NO_HOLDER;
-      g_inode_sem.count  = 0;
-      nxsem_post(&g_inode_sem.sem);
-    }
+  nxrmutex_unlock(&g_inode_lock);
```

Comment:
```suggestion
  DEBUGVERIFY(nxrmutex_unlock(&g_inode_lock));
```
This is to get back `DEBUGASSERT(g_inode_sem.holder == getpid());`